sqldelight-compiler/src/main/antlr/com/squareup/sqldelight/Sqlite.g4

```diff
@@ -390,11 +390,12 @@ compound_operator
  ;
 
 signed_number
- : ( '+' | '-' )? NUMERIC_LITERAL
+ : ( '+' | '-' )? ( INTEGER_LITERAL | REAL_LITERAL )
```

Comment:
its specific for us (see https://www.sqlite.org/syntax/numeric-literal.html). It's really just for the java-side precision since sqlite. For whatever reason sqlite makes the discrepancy at this level between storage type (NUMERIC) and data type (INTEGER / REAL)